Convert Gradebook to frontend-base

holaontiveros opened the pull request:
**TL;DR -** 
Conversion to use frontend-base and match other apps

Fixes #497 

**What changed?**

- Added dependencies required for frontend-base and recreated the lock so it works properly
- Changed the base structure to match the compatible one for frontend-base 
- Changed all imports to be relative so it's compatible
- Updated all references from `@edx/frontend-platform***` to `@openedx/frontend-base`

**Testing Instructions**
Assuming you already have everything setup in your local to run frontend-base apps

- make sure you are in the branch for this PR
- clean all your node_modules `rm -rf node_modules` and reinstall them npm ci
- run the dev server and verify you are able to access it `npm run dev`

The app should be working but given the conversations with @arbrandes the tests are going to wait until we are able to migrate react-tes-utils in master and rebase them


Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality


FYI: @openedx/content-aurora

brian-smith-tcril committed:
feat!: add design tokens support (#470)

BREAKING CHANGE: Pre-design-tokens theming is no longer supported.

Co-authored-by: Diana Olarte <diana.olarte@edunext.co>

edx-requirements-bot committed:
chore: update browserslist DB (#480)

Co-authored-by: jansenk <1639231+jansenk@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump dawidd6/action-send-mail from 4 to 6

Bumps [dawidd6/action-send-mail](https://github.com/dawidd6/action-send-mail) from 4 to 6.
- [Release notes](https://github.com/dawidd6/action-send-mail/releases)
- [Commits](https://github.com/dawidd6/action-send-mail/compare/v4...v6)

---
updated-dependencies:
- dependency-name: dawidd6/action-send-mail
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

edx-requirements-bot committed:
chore: update browserslist DB (#482)

Co-authored-by: jansenk <1639231+jansenk@users.noreply.github.com>

kdmccormick committed:
chore: Delete CODEOWNERS (#493)

See: https://github.com/openedx/axim-engineering/issues/1511

edx-requirements-bot committed:
chore: update browserslist DB (#495)

Co-authored-by: jansenk <1639231+jansenk@users.noreply.github.com>

holaontiveros committed:
chore: update dependency related things

holaontiveros committed:
chore: clena gitignore

holaontiveros committed:
chore: add main ts configs

holaontiveros committed:
chore: update main configs related with testing

holaontiveros committed:
chore: update eslint rules

holaontiveros committed:
chore: remove old webpack config files

holaontiveros committed:
chore: update i18n related things

holaontiveros committed:
chore: update getConfig

holaontiveros committed:
chore: replace all imports from @edx/frontend-platform and small lint / path fixes on files that had them

holaontiveros committed:
chore: delete .env and add new site.config and related files

holaontiveros committed:
chore: remove initialization

holaontiveros committed:
chore: migrate / remove header footer dependencies

holaontiveros committed:
chore: export the modules of your app in your index.ts file

holaontiveros committed:
chore: app.scss updates

holaontiveros committed:
chore: remove use of process.env

holaontiveros committed:
chore: convert @import to @use in SCSS files

holaontiveros committed:
chore: refactor slots

holaontiveros committed:
chore: remove build from CI

holaontiveros committed:
chore: updated import paths

holaontiveros committed:
chore: update main component for better compatibility (HOC to hooks)

holaontiveros committed:
chore: main renames to match other apps

holaontiveros committed:
chore:  updated fa icons to prevent problems of icons not showing

holaontiveros committed:
chore: remove undesired features

openedx-webhooks commented:
Thanks for the pull request, @holaontiveros!


This repository is currently maintained by `@farhaanbukhsh`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>

<summary>:radio_button: Update the status of your PR</summary>

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.
</details>
<!-- comment:end_of_wip -->


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IHRydWV9 -->

holaontiveros committed:
chore: small readme updates

diana-villalvazo-wgu commented on package.json:
```diff
@@ -51,33 +71,14 @@
-  "devDependencies": {
-    "@edx/browserslist-config": "^1.1.1",
-    "@openedx/frontend-build": "^14.3.3",
-    "@testing-library/react": "^16.2.0",
```
should we delete this one?

arbrandes reviewed (changes requested):
Thank you for all the hard work!  There's a lot to like, so apologies in advance for focusing on the problems.

In addition to the inline comments that focus on the code, these are the issues I found when running this, so far:

* The "Back to Dashboard" breadcrumb has its URL broken.  Note the "undefined": `http://apps.local.openedx.io:1994/gradebook/undefined/courses/course-v1:Axim+DC01+dev/instructor`
* The header is not the same: the logo doesn't link back to the dashboard, there is no "Courses" link (which also links back to the dashboard), and the user menu is missing its content (fix in the inline comments)
* The page is missing the subtitle that displays the course ID ("course-v1:OpenedX+DemoX+DemoCourse")
* The search box is missing the placeholder "Search for a learner"

Otherwise, as mentioned above, things seem to be working just fine!

holaontiveros committed:
chore: fixed comments

holaontiveros commented:
> Thank you for all the hard work! There's a lot to like, so apologies in advance for focusing on the problems.
> 
> In addition to the inline comments that focus on the code, these are the issues I found when running this, so far:
> 
> * The "Back to Dashboard" breadcrumb has its URL broken.  Note the "undefined": `http://apps.local.openedx.io:1994/gradebook/undefined/courses/course-v1:Axim+DC01+dev/instructor`
> * The header is not the same: the logo doesn't link back to the dashboard, there is no "Courses" link (which also links back to the dashboard), and the user menu is missing its content (fix in the inline comments)
> * The page is missing the subtitle that displays the course ID ("course-v1:OpenedX+DemoX+DemoCourse")
> * The search box is missing the placeholder "Search for a learner"
> 
> Otherwise, as mentioned above, things seem to be working just fine!

- breadcrumbs url fixed one LMS_BASE_URL escaped from the first round
- added slot for courses link, fixed roles for externalURls
- fixed typo on route param `:id` -> `:courseid`
- about the label there seems to be a mismatch in params from paragon v22 and v23, the component seems to be using `inputLabel` which is expected on paragon v23 but in 22 it's just `label` now the issue with that it's that it has a too long text for a label in v22 I think...

<img width="483" height="181" alt="image" src="https://github.com/user-attachments/assets/e23a0c65-1471-4f9a-8d44-de504447a488" />

So we either use a placeholder for now or we play around with some CSS to override some default behaviours @arbrandes

holaontiveros committed:
chore: comment fixes

holaontiveros committed:
chore: added right URL for courses menu item

jacobo-dominguez-wgu commented on tsconfig.json:
```diff
@@ -0,0 +1,15 @@
+    "src/**/*",
+    "app.d.ts",
+    "babel.config.js",
+    ".eslintrc.js",
```
repalce it with `eslint.config.js`

holaontiveros committed:
chore: fixed eslint config name

mphilbrick211 commented:
> Hi @arbrandes @holaontiveros - is this still in progress?

Friendly ping on this @holaontiveros @arbrandes

arbrandes commented:
Let's close this until we get back here for Willow (though I suspect we'll start it from scratch).
